core/clustersmngr/middleware.go

```diff
@@ -0,0 +1,50 @@
+package clustersmngr
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/weaveworks/weave-gitops/pkg/server/auth"
+)
+
+// WithClustersClients creates clusters client for provided user in the context
+func WithClustersClients(clustersFetcher ClusterFetcher, next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		user := auth.Principal(r.Context())
+		if user == nil {
+			next.ServeHTTP(w, r)
+			return
+		}
+
+		clusters, err := clustersFetcher.Fetch(r.Context())
```

Comment:
we are punting the caching of the available clusters to another PR right?

tho i suppose the implementor of the interface can cache on that side. actually that is the better idea, nvm!

Reply:
yep, not handling any of that now.